File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Views/Pages/PrivacyPolicyView.swift
import SwiftUI

struct PrivacyPolicyView: View {
    @Environment(\.presentationMode) var presentationMode
    @State private var showDeleteAlert = false
    @State private var showDeactivateAlert = false

    var body: some View {
        VStack(spacing: 0) {
            // Header
            HStack {
                Button(action: {
                    presentationMode.wrappedValue.dismiss()
                }) {
                    Image(systemName: "arrow.left")
                        .font(.system(size: 20, weight: .medium))
                        .foregroundColor(Color(hex: "#1F2937"))
                }

                Spacer()

                Text("Privacy Center")
                    .font(.system(size: 18, weight: .bold))
                    .foregroundColor(Color(hex: "#1F2937"))

                Spacer()

                Color.clear.frame(width: 20, height: 20)
            }
            .padding(.horizontal, 16)
            .padding(.vertical, 14)
            .background(Color.white)
            .overlay(
                Rectangle()
                    .fill(Color(hex: "#E5E7EB"))
                    .frame(height: 1),
                alignment: .bottom
            )

            ScrollView {
                VStack(spacing: 0) {
                    Group {
                        menuItem(title: "Privacy Policy", icon: "doc.text.fill")
                        Divider().padding(.leading, 16)
                        menuItem(title: "Request My Data", icon: "arrow.down.doc.fill")
                        Divider().padding(.leading, 16)
                        menuItem(title: "Consent Management", icon: "checkmark.shield.fill")
                        Divider().padding(.leading, 16)
                        menuItem(title: "Grievance Redressal", icon: "exclamationmark.bubble.fill")
                    }
                    .background(Color.white)

                    Spacer().frame(height: 24)

                    VStack(spacing: 0) {
                        Button(action: { showDeactivateAlert = true }) {
                            HStack {
                                Text("Deactivate Account")
                                    .foregroundColor(.blue)
                                Spacer()
                                Image(systemName: "chevron.right")
                                    .foregroundColor(.gray)
                                    .font(.system(size: 14))
                            }
                            .padding()
                            .background(Color.white)
                        }

                        Divider().padding(.leading, 16)

                        Button(action: { showDeleteAlert = true }) {
                            HStack {
                                Text("Delete Account")
                                    .foregroundColor(.red)
                                Spacer()
                                Image(systemName: "chevron.right")
                                    .foregroundColor(.gray)
                                    .font(.system(size: 14))
                            }
                            .padding()
                            .background(Color.white)
                        }
                    }
                    .background(Color.white)
                    .cornerRadius(8)
                    .padding(.horizontal, 0)  // Full width group
                }
                .padding(.vertical, 16)
            }
            .background(Color(hex: "#F3F4F6"))
        }
        .navigationBarHidden(true)
        .alert("Deactivate Account", isPresented: $showDeactivateAlert) {
            Button("Cancel", role: .cancel) {}
            Button("Deactivate", role: .destructive) {
                // Handle deactivation
            }
        } message: {
            Text("Are you sure you want to deactivate your account? This is temporary.")
        }
        .alert("Delete Account", isPresented: $showDeleteAlert) {
            Button("Cancel", role: .cancel) {}
            Button("Delete", role: .destructive) {
                // Handle deletion
            }
        } message: {
            Text("Are you sure you want to delete your account? This action cannot be undone.")
        }
    }

    private func menuItem(title: String, icon: String) -> some View {
        HStack {
            Image(systemName: icon)
                .foregroundColor(Color(hex: "#6B7280"))
                .frame(width: 24)
            Text(title)
                .foregroundColor(Color(hex: "#374151"))
            Spacer()
            Image(systemName: "chevron.right")
                .foregroundColor(.gray)
                .font(.system(size: 14))
        }
        .padding()
        .background(Color.white)
    }
}

File: apps/Applications/android/app/src/main/java/com/ecommerceearn/app/ui/pages/PrivacyPolicyView.kt
package com.ecommerceearn.app.ui.pages

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.ecommerceearn.app.data.manager.NavigationManager

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun PrivacyPolicyView() {
    var showDeactivateAlert by remember { mutableStateOf(false) }
    var showDeleteAlert by remember { mutableStateOf(false) }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Privacy Center", fontSize = 18.sp, fontWeight = FontWeight.Bold, color = Color(0xFF1F2937)) },
                navigationIcon = {
                    IconButton(onClick = { NavigationManager.goBack() }) {
                        Icon(Icons.Default.ArrowBack, contentDescription = "Back", tint = Color(0xFF1F2937))
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(containerColor = Color.White)
            )
        },
        containerColor = Color(0xFFF3F4F6)
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .verticalScroll(rememberScrollState())
        ) {
            Spacer(modifier = Modifier.height(16.dp))

            // White group
            Column(modifier = Modifier.background(Color.White).fillMaxWidth()) {
                PrivacyMenuItem("Privacy Policy", Icons.Default.Description)
                Divider(modifier = Modifier.padding(start = 16.dp))
                PrivacyMenuItem("Request My Data", Icons.Default.FileDownload)
                Divider(modifier = Modifier.padding(start = 16.dp))
                PrivacyMenuItem("Consent Management", Icons.Default.Shield)
                Divider(modifier = Modifier.padding(start = 16.dp))
                PrivacyMenuItem("Grievance Redressal", Icons.Default.Warning)
            }

            Spacer(modifier = Modifier.height(24.dp))

            // Danger group
            Column(modifier = Modifier.background(Color.White).fillMaxWidth()) {
                Row(
                    modifier = Modifier
                        .clickable { showDeactivateAlert = true }
                        .fillMaxWidth()
                        .padding(16.dp),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text("Deactivate Account", color = Color.Blue, fontSize = 16.sp)
                    Icon(Icons.Default.ChevronRight, contentDescription = null, tint = Color.Gray)
                }

                Divider(modifier = Modifier.padding(start = 16.dp))

                Row(
                    modifier = Modifier
                        .clickable { showDeleteAlert = true }
                        .fillMaxWidth()
                        .padding(16.dp),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text("Delete Account", color = Color.Red, fontSize = 16.sp)
                    Icon(Icons.Default.ChevronRight, contentDescription = null, tint = Color.Gray)
                }
            }
        }

        if (showDeactivateAlert) {
            AlertDialog(
                onDismissRequest = { showDeactivateAlert = false },
                title = { Text("Deactivate Account") },
                text = { Text("Are you sure you want to deactivate your account? This is temporary.") },
                confirmButton = {
                    TextButton(onClick = { showDeactivateAlert = false }) { Text("Deactivate", color = Color.Red) }
                },
                dismissButton = {
                    TextButton(onClick = { showDeactivateAlert = false }) { Text("Cancel") }
                }
            )
        }

        if (showDeleteAlert) {
            AlertDialog(
                onDismissRequest = { showDeleteAlert = false },
                title = { Text("Delete Account") },
                text = { Text("Are you sure you want to delete your account? This action cannot be undone.") },
                confirmButton = {
                    TextButton(onClick = { showDeleteAlert = false }) { Text("Delete", color = Color.Red) }
                },
                dismissButton = {
                    TextButton(onClick = { showDeleteAlert = false }) { Text("Cancel") }
                }
            )
        }
    }
}

@Composable
fun PrivacyMenuItem(title: String, icon: ImageVector) {
    Row(
        modifier = Modifier
            .clickable { /* no-op in stub */ }
            .fillMaxWidth()
            .padding(16.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(icon, contentDescription = null, tint = Color(0xFF6B7280), modifier = Modifier.size(24.dp))
        Spacer(modifier = Modifier.width(16.dp))
        Text(title, color = Color(0xFF374151), fontSize = 16.sp, modifier = Modifier.weight(1f))
        Icon(Icons.Default.ChevronRight, contentDescription = null, tint = Color.Gray)
    }
}
